File: C#/프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.cs
using System;
using System.Linq;
using System.Collections.Generic;

public class Solution {
    public int solution(int[] array) {
        // Dictionary 생성
        Dictionary<int, int> counts = new Dictionary<int, int>();
        
        // foreach로 숫자 갯수 카운트
        foreach(var a in array) {
            if (!counts.ContainsKey(a))
                counts.Add(a, 0);
            counts[a]++;
        }
        
        // 기본값 초기화
        int answer = -1;
        int maxValue = 0;
        
        // foreach로 최빈값 탐색
        foreach(var c in counts) {
            // 같은 카운트의 수가 있으면 -1
            if (c.Value == maxValue) answer = -1;
            // maxValue 찾기
            if (c.Value > maxValue) {
                maxValue = c.Value;
                answer = c.Key;
            }
        }
        
        return answer;
    }
}

File: Kotlin/프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.kt
class Solution {
    fun solution(array: IntArray): Int {
        // 숫자별로 그룹화 하고, 그룹 사이즈를 Map의 value로 반환
        val counts = array.groupBy{ it }
                          .mapValues { it.value.size }
        
        // MaxCount는 Map의 Value중 max 값
        var maxCount: Int = counts.values.maxOrNull() ?: 0
        // List로 필터링
        val answer = counts.filterValues { it == maxCount }
                           .keys
                           .toList()
        
        // List의 길이가 2개(최댓값으로 필터링한 값)
        if (answer.size > 1) return -1
        // 아니면 첫번째 값(1개밖에 없음)
        return answer[0]
    }
}